nix/modules/raid-config.nix

```diff
@@ -72,9 +62,14 @@ in
         type = "mdadm";
         level = 1;
         content = {
-          type = "filesystem";
-          format = "ext4";
-          mountpoint = "/";
+          type = "luks";
+          name = "root-encrypted";
+          settings.keyFile = "/run/diskKeyFile";
```

Comment:
this one also needs pregeneration, but only during the initial disko creation phase. the manager then needs to provide that secret during the initrd booting